tests/functional/test_configuration.py

```diff
@@ -1,35 +1,36 @@
 """Tests for the config command
 """
-
 import re
 import textwrap
 
 import pytest
 
 from pip._internal.cli.status_codes import ERROR
 from pip._internal.configuration import CONFIG_BASENAME, get_configuration_files
+from tests.lib import PipTestEnvironment
 from tests.lib.configuration_helpers import ConfigurationMixin, kinds
+from tests.lib.venv import VirtualEnvironment
 
 
-def test_no_options_passed_should_error(script):
+def test_no_options_passed_should_error(script: PipTestEnvironment) -> None:
     result = script.pip("config", expect_error=True)
     assert result.returncode == ERROR
 
 
 class TestBasicLoading(ConfigurationMixin):
     @pytest.mark.skip("Can't modify underlying file for any mode")
-    def test_reads_file_appropriately(self, script):
+    def test_reads_file_appropriately(self, script: PipTestEnvironment) -> None:
         contents = """
             [test]
             hello = 1
         """
 
-        with self.patched_file(kinds.USER, contents):
+        with self.patched_file(kinds.USER, contents):  # type: ignore[attr-defined]
```

Comment:
Is this needed for `kinds.USER`?

Reply:
This test is unconditionally skipped, so it doesn't present an issue on test a run.

However, this is referencing a method that doesn't exist, `patched_file`.

It looks like it was removed or renamed in bd850c079851d5458e4af82921739666545b6170.

Here is the actual error:

```
tests/functional/test_configuration.py:28: error: "TestBasicLoading" has no
attribute "patched_file"  [attr-defined]
            with self.patched_file(kinds.USER, contents):
                 ^
```

Other options:

1. Remove this test as it doesn't run anwyay.
2. Fix this test to run. This requires details from someone that knows what the test _should_ be doing. It was added in 6cf5d7db54ee272fa9af66d45a504d5994693ae4.

Reply:
FWIW, here is a grep of tests that use `@pytest.mark.skip` without a boolean check:

```
tests/functional/test_search.py
70   @pytest.mark.skip(

tests/functional/test_configuration.py
20       @pytest.mark.skip("Can't modify underlying file for any mode")

tests/functional/test_install.py
557  @pytest.mark.skip("Fails on new resolver")
```

Reply:
The search tests are… well, you know the story about `pip search`. The new resolver one is more like a todo, we should decide what to do with it, either rewrite or drop it completely.

I’m not familiar with the configuration one either, maybe @pradyunsg remembers

https://github.com/pypa/pip/blame/09103e83e35dc6ba95639710148c100498010dfe/tests/functional/test_configuration.py#L20

Reply:
Yea, just drop the configuration one. It was aspirational at best. :)

Reply:
> Yea, just drop the configuration one. It was aspirational at best. :)

Done in https://github.com/pypa/pip/pull/10445.